packages/salesforcedx-vscode-apex/src/languageServerOrphanHandler.ts

```diff
@@ -0,0 +1,116 @@
+      for (const processInfo of orphanedProcesses) {
+        try {
+          await terminateProcess(processInfo.pid);
+          telemetryService.sendEventData('apexLSPStartup', undefined, { terminateSuccessful: 1 });
+          showProcessTerminated(processInfo);
+        } catch (err) {
+          showTerminationFailed(processInfo, err);
+          telemetryService.sendEventData(
+            'apexLSPStartup',
+            { terminationErrorMessage: typeof err === 'string' ? err : err?.message ? err.message : 'unknown' },
+            { terminateSuccessful: 0 });
+        }
+      }
+    } else {
+      telemetryService.sendEventData('apexLSPStartup', undefined, { orphanCount: orphanedProcesses.length, didTerminate: 0 });
+    }
+  }
+}
+
+/**
+ * Ask the user how to resolve found orphaned language server instances
+ * @param orphanedProcesses
+ * @returns boolean
+ */
+export async function getResolutionForOrphanProcesses(orphanedProcesses: ProcessDetail[]): Promise<boolean> {
+  const orphanedCount = orphanedProcesses.length;
+
+  if (orphanedCount === 0) {
+    return false;
+  }
+
+  let choice: string | undefined = nls.localize(SHOW_PROCESSES);
+  do {
+    choice = await vscode.window.showWarningMessage(
+      nls.localize(
+        TERMINATE_ORPHANGED_PROCESSES,
+        orphanedCount
+      ),
+      TERMINATE_PROCESSES_BTN,
+      SHOW_PROCESSES_BTN
+    ) ?? DISMISSED_DEFAULT;
+
+    if (requestsTermination(choice) && await terminationConfirmation(orphanedCount)) {
+      return true;
+    } else if (showProcesses(choice)) {
+      const processId: string = PROCESS_ID;
+      const parentProcessId: string = PROCESS_PARENT_ID;
+      const processCommand: string = COMMAND;
+      const title = `${processId} ${parentProcessId} ${processCommand}`;
+      const titleUnderline = `${'='.repeat(processId.length)} ${'='.repeat(parentProcessId.length)} ${'='.repeat(processCommand.length)}`;
+      const processList = orphanedProcesses.map(processInfo => {
+        return `${processInfo.pid.toString().padStart(processId.length)} ${processInfo.ppid.toString().padStart(parentProcessId.length)} ${processInfo.command}`;
+      });
+      channelService.showChannelOutput();
+      channelService.appendLine([ADVICE, '', title, titleUnderline, ...processList].join('\n'));
+    }
+  } while (!choice || showProcesses(choice));
+  return false;
+}
+
+async function terminationConfirmation(orphanedCount: number): Promise<boolean> {
+  const choice = await vscode.window.showWarningMessage(
+    nls.localize(
+      CONFIRM,
+      orphanedCount
+    ),
+    YES,
+    CANCEL
+  );
+  return choice === YES;
+}
+
+function requestsTermination(choice: string | undefined): boolean {
+  return choice === TERMINATE_PROCESSES_BTN;
+}
+
+function showProcesses(choice: string) {
+  return choice === SHOW_PROCESSES_BTN;
+}
+
+function showProcessTerminated(processDetal: ProcessDetail) {
```

Comment:
nit: processDelta (I think)

Reply:
is `processDetal` a typo?

Reply:
Ah, thanks! In my  CS 101 class someone asked, "What should I name things?" and the instructor said "I don't care, call it "dog" if you want, just be sure to call it "dog" everywhere you want to use it. My typing skills take full advantage of that principal.